src/plugins/github/fetchGitHubRepo.js

```diff
@@ -42,111 +45,100 @@ export default function fetchGitHubRepo(
     throw new Error(`Invalid token: ${token}`);
   }
 
-  const query = `query FetchData($repoOwner: String!, $repoName: String!) {
-    repository(owner: $repoOwner, name: $repoName) {
-      issues(first: 100) {
-        pageInfo {
-          hasNextPage
-        }
-        nodes {
-          id
-          title
-          body
-          number
-          author {
-            ...whoami
-          }
-          comments(first: 20) {
-            pageInfo {
-              hasNextPage
-            }
-            nodes {
-              id
-              author {
-                ...whoami
-              }
-              body
-              url
-            }
-          }
-        }
-      }
-      pullRequests(first: 100) {
-        pageInfo {
-          hasNextPage
-        }
-        nodes {
-          id
-          title
-          body
-          number
-          author {
-            ...whoami
-          }
-          comments(first: 20) {
-            pageInfo {
-              hasNextPage
-            }
-            nodes {
-              id
-              author {
-                ...whoami
-              }
-              body
-              url
-            }
-          }
-          reviews(first: 10) {
-            pageInfo {
-              hasNextPage
-            }
-            nodes {
-              id
-              body
-              author {
-                ...whoami
-              }
-              state
-              comments(first: 10) {
-                pageInfo {
-                  hasNextPage
-                }
-                nodes {
-                  id
-                  body
-                  author {
-                    ...whoami
-                  }
-                }
-              }
-            }
-          }
-        }
-      }
-    }
-  }
-
-  fragment whoami on Actor {
-    __typename
-    login
-    ... on User {
-      id
-    }
-    ... on Organization {
-      id
-    }
-    ... on Bot {
-      id
-    }
-  }
-  `;
+  const query = GraphQLQueries.stringify.body(
+    createQuery(),
+    GraphQLQueries.inlineLayout()
+  );
   const variables = {repoOwner, repoName};
   const payload = {query, variables};
   return postQuery(payload, token);
 }
 
 const GITHUB_GRAPHQL_SERVER = "https://api.github.com/graphql";
 
+function createQuery(): Body {
+  const b = GraphQLQueries.build;
+  const makePageInfo = () => b.field("pageInfo", {}, [b.field("hasNextPage")]);
+  const makeAuthor = () => b.field("author", {}, [b.fragmentSpread("whoami")]);
+  const body: Body = [
+    b.query(
+      "FetchData",
+      [b.param("repoOwner", "String!"), b.param("repoName", "String!")],
+      [
+        b.field(
+          "repository",
+          {owner: b.variable("repoOwner"), name: b.variable("repoName")},
+          [
+            b.field("issues", {first: b.literal(100)}, [
```

Comment:
Maybe factor all the first (N) arguments into constants?

Reply:
I'd like to keep the query identical in this commit. We can do this after, if you like.